ios/zxing/iphone/ZXingWidget/Classes/OverlayView.m

```diff
@@ -42,44 +42,73 @@ - (id) initWithFrame:(CGRect)theFrame
     self = [super initWithFrame:theFrame];
     if( self ) {
         rectangleEnabled = isRectangleEnabled;
-        CGFloat rectSize = self.frame.size.width - kPadding * 2;
-        if (!oneDMode) {
-            cropRect = CGRectMake(kPadding, (self.frame.size.height - rectSize) / 2, rectSize, rectSize);
-        } else {
-            CGFloat rectSize2 = self.frame.size.height - kPadding * 2;
-            cropRect = CGRectMake(kPadding, kPadding, rectSize, rectSize2);		
-        }
-        
         self.backgroundColor = [UIColor clearColor];
         self.oneDMode = isOneDModeEnabled;
-        if (isCancelEnabled) {
+        
+        if (isCancelEnabled)
+        {
             UIButton *butt = [UIButton buttonWithType:UIButtonTypeRoundedRect]; 
             self.cancelButton = butt;
             [cancelButton setTitle:@"Cancel" forState:UIControlStateNormal];
-            if (oneDMode) {
-                [cancelButton setTransform:CGAffineTransformMakeRotation(M_PI/2)];
-                
-                [cancelButton setFrame:CGRectMake(20, 175, 45, 130)];
-            }
-            else {
-                CGSize theSize = CGSizeMake(100, 50);
-                CGRect theRect = CGRectMake((theFrame.size.width - theSize.width) / 2, cropRect.origin.y + cropRect.size.height + 20, theSize.width, theSize.height);
-                [cancelButton setFrame:theRect];
-                
-            }
-            
             [cancelButton addTarget:self action:@selector(cancel:) forControlEvents:UIControlEventTouchUpInside];
             [self addSubview:cancelButton];
             [self addSubview:imageView];
         }
         
-        if (overlay != nil) {
+        [self updateViewsWithFrame:theFrame];
+        
+        if (overlay != nil)
+        {
             [self addSubview:overlay];
         }
     }
     return self;
 }
 
+-(void)updateViewsWithFrame:(CGRect)newFrame
```

Comment:
I have not updated the library. I made modification in that.
 I think the Zingwidget library which we are using is either from- https://github.com/anurlybayev/zxing-iphone/tree/master/ZXingWidget     or 
 from  https://github.com/zxing/zxing/tree/00f634024ceeee591f54e6984ea7dd666fab22ae/iphone/ZXingWidget .
 Both are updated 4years back only. I think there is no active work on these libraries. So we can modify UI part of these libraries as per our requirement. 
Or we should use some active library, e.g https://github.com/TheLevelUp/ZXingObjC, where we can create our own UI as given in there example - https://github.com/TheLevelUp/ZXingObjC/tree/master/examples/BarcodeScanner.

@hansemannn @eric34  Thoughts ?